vision/tests/unittests/test_image_classification.py

```diff
@@ -1,19 +1,44 @@
 from autogluon.vision import ImagePredictor as Task
+import pandas as pd
+import copy
 
 def test_task():
     dataset, _, test_dataset = Task.Dataset.from_folders('https://autogluon.s3.amazonaws.com/datasets/shopee-iet.zip')
+    dataset_copy = copy.deepcopy(dataset)
     model_list = Task.list_models()
     classifier = Task()
     classifier.fit(dataset, num_trials=2, hyperparameters={'epochs': 1, 'early_stop_patience': 3})
+    # assert input dataset not altered
+    assert all(dataset_copy.index == dataset.index)
     test_result = classifier.predict(test_dataset)
     single_test = classifier.predict(test_dataset.iloc[0]['image'])
     single_proba = classifier.predict_proba(test_dataset.iloc[0]['image'])
-    print('test result', test_result)
     classifier.save('classifier.ag')
     classifier2 = Task.load('classifier.ag')
     fit_summary = classifier2.fit_summary()
     test_acc = classifier2.evaluate(test_dataset)
     test_proba = classifier2.predict_proba(test_dataset)
     test_feature = classifier2.predict_feature(test_dataset)
     single_test2 = classifier2.predict(test_dataset.iloc[0]['image'])
+    assert isinstance(single_test2, pd.Series)
     assert single_test2.equals(single_test)
+    # to numpy
```

Comment:
test equivalence with pandas variants via `pandas_variant.to_numpy()`

Refer to https://numpy.org/doc/stable/reference/generated/numpy.array_equal.html